server.js

```diff
@@ -4739,6 +4739,53 @@ cache(function(data, match, sendBadge, request) {
   }
 }));
 
+// Issue Stats integration.
+camp.route(/^\/issuestats\/([^\/]+)\/(.+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var host = match[1];      // e.g. `github`
+  var userRepo = match[2];  // e.g. `ruby/rails`
+  var type = match[3];      // e.g. `issue` or `pr`
+  var format = match[4];
+
+  var badgeData = getBadgeData('Issue Stats', data);
+
+  var url = 'http://issuestats.com/' + host + '/' + userRepo;
+  var qs = {format: 'json'};
+  if (data.hasOwnProperty('concise')) {
+    qs.concise = data.concise;
+  }
+  var options = {
+    method: 'GET',
+    url: url,
+    qs: qs,
+    gzip: true,
+    json: true
+  };
+  request(options, function(err, res, json) {
+    if (err != null || res.statusCode >= 500) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    if (res.statusCode >= 400 || !json || typeof json !== 'object') {
+      badgeData.text[1] = 'not found';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    var label = json[type + '_badge_preamble'];
+    var value = json[type + '_badge_words'];
+    var color = json[type + '_badge_color'];
+
+    if (label) badgeData.text[0] = label;
+    badgeData.text[1] = value || 'invalid';
+    if (color) badgeData.colorscheme = color;
+
+    sendBadge(format, badgeData);
```

Comment:
If `sendBadge` throws, what should I do in the catch block?  It appears that other badges call `sendBadge` with `text[1] = 'invalid'`, but wouldn't it just throw again?  I can certainly do that, I'm just trying to understand what should be happening.